internal/dinosaur/pkg/clusters/ocm_provider.go

```diff
@@ -124,7 +125,7 @@ func (o *OCMProvider) ApplyResources(clusterSpec *types.ClusterSpec, resources t
 	if err != nil {
 		svcErr := svcErrors.ToServiceError(err)
 		if !svcErr.Is404() {
-			return nil, err
+			return nil, fmt.Errorf("applying resources: %w", svcErr)
```

Comment:
For me `"applying resources"` would be a better wrapping for errors at the caller site of `ApplyResources`. Here I would suggest something like 
```suggestion
			return nil, fmt.Errorf("retrieving sync set %s for cluster %s: %w", resources.Name, clusterSpec.InternalID, svcErr)
```